host-controller/src/main/java/org/jboss/as/domain/controller/operations/coordination/OperationSlaveStepHandler.java

```diff
@@ -135,6 +135,19 @@ HostControllerExecutionSupport addSteps(final OperationContext context, final Mo
                 resolver, hostControllerExecutionSupport, originalAddress, originalRegistration, multiPhaseLocalContext);
         context.addStep(sorh, OperationContext.Stage.DOMAIN);
 
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+
+                context.completeStep(new OperationContext.ResultHandler() {
+                    @Override
+                    public void handleResult(OperationContext.ResultAction resultAction, OperationContext context, ModelNode operation) {
+                        hostControllerExecutionSupport.complete(context, resultAction == OperationContext.ResultAction.ROLLBACK);
```

Comment:
I believe this can be moved to the existing ResultHandler passed to completeStep by OperationSlaveStepHandler's execute method.